avro-serializer/src/test/java/io/confluent/kafka/serializers/KafkaAvroSerializerTest.java

```diff
@@ -613,6 +649,49 @@ public void testKafkaAvroSerializerReflectionRecord() {
     assertEquals(widget, obj);
   }
 
+  @Test
+  public void testKafkaAvroSerializerReflectionRecordWithNullField() {
+    byte[] bytes;
+    Object obj;
+
+    ExtendedWidget widget = new ExtendedWidget();
+    // intentionally miss setting Age field
+    widget.setName("alice");
+    Schema schema = ReflectData.AllowNull.get().getSchema(widget.getClass());
+
+    try {
+      reflectionAvroSerializer.serialize(topic, widget);
+      fail("Sending instance with null field should fail reflection serializer");
+    } catch (SerializationException e){
+      //this is expected
+    }
+
+    Map configs = ImmutableMap.of(
+        KafkaAvroDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG, "bogus",
+        AbstractKafkaSchemaSerDeConfig.SCHEMA_REFLECTION_CONFIG, true,
+        KafkaAvroSerializerConfig.AVRO_REFLECTION_ALLOW_NULL_CONFIG, true
+    );
+    reflectionAvroDeserializer.configure(configs, false);
+    reflectionAvroSerializer.configure(configs, false);
+
+    bytes = reflectionAvroSerializer.serialize(topic, widget);
+    obj = reflectionAvroDecoder.fromBytes(bytes, schema);
+    assertTrue(
+        "Returned object should be a io.confluent.kafka.example.User",
+        ExtendedWidget.class.isInstance(obj)
+    );
+    assertEquals(widget, obj);
+
+    obj = reflectionAvroDeserializer.deserialize(topic, bytes, schema);
+    //obj = reflectionAvroDeserializer.deserialize(topic, bytes);
```

Comment:
Are this two lines interchangeable? The test pass for both of them.

Reply:
Let's delete 684 and uncomment 685